### ISSUE TITLE
lightbox-gallery does not show description/X overlay on Firefox 63 and 65 (64 is fine!)

### ISSUE THREAD

**Issue by IamAlta** [ampproject/amphtml#20586]
## What's the issue?
Feature request for amp-carousel with 1 image.

I read somewhere that lightbox is being depreciated, so I am using amp-carousel to show a normal size image with an extra large popup image. I like the way the amp-carousel looks and feels better than lightbox.

The AMP page has one or more images, and I can program amp-carousel okay with >1 image.
The site is dynamic and sometimes there is 1 image, or 2, 3, .., images, 
so whether there is one or more images, it uses amp-carousel to show the popup extra large image.

The page starts with a standard smaller image.
When that is clicked the amp-carousel is launched and shows larger images in the amp-carousel.

BUT, when there is only one image and the standard image is clicked, the carousel shows the enlarged image okay but there is no way to close it and get back to the page.

Can there be an "X" on it to close it?

## How do we reproduce the issue?
See here
https://codepen.io/iamalta/pen/KrvBgj

## What browsers are affected?
Chrome, Firefox

## Which AMP version is affected?
Powered by AMP ⚡ HTML – Version 1901241837330

Thank you for your help!

**Comment by torch2424** [comment]
Triaging to @aghassemi 😄 Feel free to re-assign.

**Comment by aghassemi** [comment]
Hi @IamAlta 

Using `amp-lightbox-gallery` component is definitely the right approach here.

As for the inline caoursel supporting one item, it is something we may consider for carousel vs (cc @nainar @sparhami) but it semantically does not makes sense and it is easy to handle on server-side like:
```
if (oneItem) {
 render <amp-img lightbox>
} else {
render <amp-carousel lightbox>
}
```

**Comment by IamAlta** [comment]
Hi @aghassemi  (cc @nainar @sparhami)

First, <amp-carousel lightbox>? 
I don't see documentation for that under amp-carousel, where I would expect it.
I did find it under amp-lightbox-gallery, but I wouldn't have looked for it there.

You have a good idea but won't quite work for option selections the way the program is written.
The number of images is unknown when the <amp-img lightbox> or <amp-carousel lightbox> is rendered.

Think of selecting an option from a list.
In a simple explanation, the program iterates over a list of product options where the lightbox is rendered, and not all options may have an image. That info, whether an image will be displayed, is not known until the option content is created.

1. Get a list of options.
2. Iterate over the options.
—2a. Create the text for current option.
—2b. Call a subroutine to render the image. The subroutine check to see if the image exists. If so, then render it. If not, skip image.
*To determine whether the option has an image, a check of whether the image exists on the server is done. (Sometimes helpers enter image names but it isn't on the server.).

There would have to be a separate iteration before step 2 loop to determine the number of images and whether or not they exist on the server.
The subroutine in 2b checks for whether the image exists on the server which results in redundant calls to the server. (This subroutine is used elsewhere, too, so that check is necessary.).

If you can modify the carousel lightbox to accept one element by just adding a close "X" to the larger image is best. Meanwhile, I'll see about editing the code on our end.

Thanks for your suggestion. Make it a wonderful day - you all earned it!

**Comment by aghassemi** [comment]
@IamAlta amp-lightbox-gallery works fine with one item and it does render a "X". You are seeing otherwise? ( I can see it working fine on https://codepen.io/iamalta/pen/KrvBgj ).

(The X and the captions can be toggle of by the user, clicking on the image shows/hides the X and the captions)

**Comment by IamAlta** [comment]
Hi @aghassemi,
I see no "X" at https://codepen.io/iamalta/pen/KrvBgj
Here are images before (left) and after (right) on FireFox Developer Edition 65.0b7 (64-bit)
![image](https://user-images.githubusercontent.com/34579958/52364067-8ff45880-2a01-11e9-945f-e516ae1f85df.png)

**Comment by aghassemi** [comment]
@IamAlta oooh might be a firefox 65 bug! FF 64 and Chrome works. Could you verify whether it works in Chrome for you?

**Comment by IamAlta** [comment]
@aghassemi,
Yes - it works on Chrome.
I just updated to FireFox Developer Edition 65.0b10 (64-bit) and still doesn't work.

**Comment by aghassemi** [comment]
@IamAlta great. Given it works on Firefox 64 and even 65 beta (not developer edition, normal beta), it is likely a bug in developer edition.

**Comment by IamAlta** [comment]
@aghassemi,
I does not work on the standard FireFox Quantum 63.0.3 (64-bit) either.

**Comment by sparhami** [comment]
This is fixed by #20516.
